### PR TITLE
notify-233: delete notifications from notifications table after they …

### PR DESCRIPTION
**Pull request by terrazoon**
…are successfully sent.

Originally I tried never writing notifications to the table unless they failed.  This was partially working but involved many code code changes and test failures and seemed to interfere with some logic about the "saved-sms", and research mode.

It occurred to me that we could just keep writing to the db as we currently do, and then delete successful notifications and move them to notification_history immediately.

### CODE REVIEW

**Review by rahearn — COMMENTED** [review]
I think I like this approach, but I believe that you're calling the new function in the incorrect spots to only remove the successful deliveries.

**Comment by terrazoon** [comment]
> I think I like this approach, but I believe that you're calling the new function in the incorrect spots to only remove the successful deliveries.

I have confirmed on my local that successful texts are deleted, but I'm not quite sure how to send unsuccessful texts.  I think Steven mentioned using an invalid phone number, but the UI is checking length and format etc so I'm not sure how to get one of those.  I tried to delete after successful retry and after success first try only, and so it's not clear to me how I would be deleting unsuccessful ones ... but I might be.

How does one test unsuccessful texts?  Is there a magic bad phone number?

**Comment by rahearn** [comment]
> How does one test unsuccessful texts?

I think you could introduce a failure by commenting out the `SNS_AWS_SECRET_ACCESS_KEY` line in `.env`. I'd expect that would leave the `notification_id` in the table and in the retry queue.

**Comment by terrazoon** [comment]
> > How does one test unsuccessful texts?
> 
> I think you could introduce a failure by commenting out the `SNS_AWS_SECRET_ACCESS_KEY` line in `.env`. I'd expect that would leave the `notification_id` in the table and in the retry queue.

I tried this but when I ran make-procfile, got this:


13:39:30 scheduler.1 | botocore.exceptions.PartialCredentialsError: Partial credentials found in explicit, missing: aws_secret_access_key

**Comment by tdlowden** [comment]
Just want to note to hold on this merge for a bit. Going to discuss more with @stvnrlly

**Comment by stvnrlly** [comment]
Just noting that we've decided to tackle SMS delivery receipts and then return to this. This will allow us to get more useful success/failure data flowing through the system and use that to determine just how this should work.

File: app/celery/provider_tasks.py [hunk 1]
```diff
@@ -37,11 +40,13 @@ def deliver_sms(self, notification_id):
                 self.retry(queue=QueueNames.RETRY, countdown=0)
             else:
                 self.retry(queue=QueueNames.RETRY)
+            insert_notification_history_delete_notifications_by_id(notification_id)
         except self.MaxRetriesExceededError:
             message = "RETRY FAILED: Max retries reached. The task send_sms_to_provider failed for notification {}. " \
                       "Notification has been updated to technical-failure".format(notification_id)
             update_notification_status_by_id(notification_id, NOTIFICATION_TECHNICAL_FAILURE)
             raise NotificationTechnicalFailureException(message)
+    insert_notification_history_delete_notifications_by_id(notification_id)
```
Comment (ccostino) [inline app/celery/provider_tasks.py:49]: Same note as the comment above.

File: app/celery/provider_tasks.py [hunk 2]
```diff
@@ -37,11 +40,13 @@ def deliver_sms(self, notification_id):
                 self.retry(queue=QueueNames.RETRY, countdown=0)
             else:
                 self.retry(queue=QueueNames.RETRY)
+            insert_notification_history_delete_notifications_by_id(notification_id)
```
[the remark this answers is not in the document]
Comment (terrazoon) [reply app/celery/provider_tasks.py]: I made this change and tested.  Looks good to me.  The only issue is that we currently don't have a way to make an sms message fail.   So I sent two successfully, waited for 5 minutes, and watched them get deleted in the db.

**Review by stvnrlly — APPROVED** [review]

File: app/celery/provider_tasks.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ccostino) [reply app/celery/provider_tasks.py]: Awesome, thanks @terrazoon!

**Pull request merged**